APHKLogicExtractor/Loader/ResourceLoader.cs

```diff
@@ -0,0 +1,138 @@
+using System.Collections.Concurrent;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using RandomizerCore.Json;
+
+namespace APHKLogicExtractor.Loader;
+
+class ResourcePathParser
+{
+    public readonly string full;
+    public readonly string[] parts;
+    public readonly string protocol;
+    public readonly string path;
+    public readonly string? additional;
+
+    public ResourcePathParser(string path)
+    {
+        this.full = path;
+        this.parts = path.Split(":");
+
+
+        if (this.parts.Length == 2)
+        {
+            if (this.parts[1].StartsWith("//"))
+            {
+                this.protocol = this.parts[0];
+                this.path = this.parts[1][2..];
+            }
+            else
+            {
+                this.protocol = "file";
+                this.path = this.parts[0];
+                this.additional = this.parts[1];
+            }
+        }
+        else if (this.parts.Length == 3)
+        {
+            this.protocol = this.parts[0];
+            this.path = this.parts[1][2..];
+            this.additional = this.parts[2];
+        }
+        else
+        {
+            this.protocol = "file";
+            this.path = this.parts[0];
+        }
+    }
+
+    public string Uri { get { return $"{this.protocol}://{this.path}"; } }
+}
+
+public class ResourceLoader(ILogger<ResourceLoader> logger)
+{
+    static readonly string CACHE_DIR = Path.Join(Path.GetTempPath(), "APLogicExtractor");
+    static readonly HttpClient client = new();
+    private ConcurrentDictionary<string, ResourceLock<byte[]?>> cache = [];
+
+    private async Task<byte[]> GetHttp(string uri)
+    {
+        // Try to grab cached remote file.
```

Comment:
That's why I save the cache to the temp directory. I don't know how it works on Windows, but on Linux it's a folder that gets erased when you reboot, so I added a cache mostly in case you need to re-run multiple tires in a row.

Reply:
Learned something new about Linux today. Unfortunately windows does not seem to have that property so we will need a plan for it.

There are a couple small things I think which might help, we could do some or all of these 
1. Log when we have loaded from the cache so any issues which might arise from using cached data would be easier to identify 
2. Add a command line argument that force-invalidates the cache
3. Check the last update time on the cached file if it exists and don't use it if it's older than some time span

Reply:
Yeah I think it should be fine, I think in theory it should be possible to do 3 but what we have now is a good start, I may revisit it later.